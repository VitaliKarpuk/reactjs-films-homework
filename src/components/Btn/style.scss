.btn__watch,
.btn__info {
  width: 199px;
  height: 60px;
  border-radius: 6px;
  margin: 5px;
  font-size: 28px;
  letter-spacing: 1px;
  line-height: 44px;
  color: #ffffff;
  text-align: center;
  margin-right: 20px;
}

.btn__watch {
  background-color: #0aaee4;
  border: none;

  &:hover {
    cursor: pointer;
  }
}

.btn__info {
  background-color: transparent;
  border: 2px solid #ffffff;

  &:hover {
    cursor: pointer;
  }
}
