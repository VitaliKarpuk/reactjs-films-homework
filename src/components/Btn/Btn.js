import React from 'react';
import PropTypes from 'prop-types';

import './style.scss';

const Btn = ({ title, className, onClick }) => {

  return (
    <button className={className} onClick={onClick}>{title}</button>
  )
}

Btn.propTypes = {
  title: PropTypes.string,
  className: PropTypes.string,
  onClick: PropTypes.func
}

export default Btn;