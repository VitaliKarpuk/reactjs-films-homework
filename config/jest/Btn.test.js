/* eslint-disable no-undef */
import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow';
import Btn from '../../src/components/Btn/Btn';


describe('Test Btn component', () => {
  test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(<Btn title='Watch Now' className="btn__watch" />);
    renderer.getRenderOutput();
    expect(result.type).toBe('button');
    expect(result.props.className).toBe('btn__watch');
    expect(result.props.children).toBe('Watch Now');
    expect(result).toMatchSnapshot();
  })
})