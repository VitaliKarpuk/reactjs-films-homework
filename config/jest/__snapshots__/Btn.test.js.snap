// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Btn component renders correctly 1`] = `
<button
  className="btn__watch"
>
  Watch Now
</button>
`;
